app/src/main/java/org/mozilla/fenix/home/topsites/TopSiteItemViewHolder.kt

```diff
@@ -72,27 +78,37 @@ class TopSiteItemViewHolder(
             binding.topSiteTitle.setCompoundDrawablesWithIntrinsicBounds(null, null, null, null)
         }
 
-        when (topSite.url) {
-            SupportUtils.POCKET_TRENDING_URL -> {
-                binding.faviconImage.setImageDrawable(getDrawable(itemView.context, R.drawable.ic_pocket))
-            }
-            SupportUtils.BAIDU_URL -> {
-                binding.faviconImage.setImageDrawable(getDrawable(itemView.context, R.drawable.ic_baidu))
-            }
-            SupportUtils.JD_URL -> {
-                binding.faviconImage.setImageDrawable(getDrawable(itemView.context, R.drawable.ic_jd))
-            }
-            SupportUtils.PDD_URL -> {
-                binding.faviconImage.setImageDrawable(getDrawable(itemView.context, R.drawable.ic_pdd))
-            }
-            SupportUtils.TC_URL -> {
-                binding.faviconImage.setImageDrawable(getDrawable(itemView.context, R.drawable.ic_tc))
-            }
-            SupportUtils.MEITUAN_URL -> {
-                binding.faviconImage.setImageDrawable(getDrawable(itemView.context, R.drawable.ic_meituan))
+        if (topSite is TopSite.Provided) {
+            CoroutineScope(IO).launch {
+                itemView.context.components.core.client.bitmapForUrl(topSite.imageUrl)?.let { bitmap ->
+                    withContext(Main) {
```

Comment:
During the context switches there is a risk of a view no longer existing when we try to call `setImageBitmap`, so we need to figure out a better way to have this as part of a lifecycle scope.